packages/kbn-config/src/env.ts

```diff
@@ -114,21 +115,11 @@ export class Env {
     this.binDir = resolve(this.homeDir, 'bin');
     this.logDir = resolve(this.homeDir, 'log');
 
-    /**
-     * BEWARE: this needs to stay roughly synchronized with the @kbn/optimizer
-     * `packages/kbn-optimizer/src/optimizer_config.ts` determines the paths
-     * that should be searched for plugins to build
-     */
-    this.pluginSearchPaths = [
```

Comment:
I decided to move this logic to `@kbn/dev-utils` because:
- the comment in `packages/kbn-optimizer/src/optimizer_config.ts` has been outdated, as a result, it's easy to have both out of sync.
- `simpleKibanaPlatformPluginDiscovery` is already in `@kbn/dev-utils`. It's better to collocate them together. Probably, we will extract this into a separate package later.

Reply:
That would cause a production dependency on `@kbn/dev-utils`, which we want to avoid.

Reply:
@spalger I will fix that. what if we extract it into a `@kbn/plugin-discovery` in the future?

Reply:
I'm not totally opposed to a plugin-discovery package, but it feels like a pretty small thing to break into its own package, I think I'd rather just consider the plugin search paths as "config" related, and then have the two discovery implementations in dev-utils and src/core.